test/DataSet.test.js

```diff
@@ -282,7 +285,111 @@ describe('DataSet', function () {
       {id: 3, content: 'Item 3'},
       {id: 4, content: 'Item 4'}
     ]);
+  });
+
+  describe('add', function () {
```

Comment:
👍 For new tests. Looks clean except for the data initialization thing.
This is stuff which is assumed to 'just work'. Which is exactly the reason why they should be regression-tested.

Reply:
Has been addressed